generators/entity-client/templates/react/src/main/webapp/app/entities/entity.reducer.ts.ejs

```diff
@@ -218,22 +218,50 @@ export const getEntity: ICrudGetAction<I<%= entityReactName %>> = id => {
   };
 };
 
-export const createEntity: ICrudPutAction<I<%= entityReactName %>> = entity => async dispatch => {
-  const result = await dispatch({
-    type: ACTION_TYPES.CREATE_<%= entityActionName %>,
-    payload: axios.post(apiUrl, cleanEntity(entity))
-  });
-  <%_ if (pagination !== 'infinite-scroll') { _%>
-  dispatch(getEntities());
-  <%_ } _%>
-  return result;
+<%
+  isEntityInMicroserviceSavingUserSnapshot =
+    applicationType === 'gateway' &&
+    locals.microserviceName &&
+    authenticationType === 'oauth2' &&
+    hasUserField &&
+    dto === 'no';
+%>
+export const createEntity: ICrudPutAction<I<%= entityReactName %>> = entity => <% if (isEntityInMicroserviceSavingUserSnapshot) { %>async (dispatch, getState) => {
+    const entityWithFullUser = {
+      ...entity,
+      user: getState().userManagement.users.filter(user => user.id === entity.user.id)[0]
```

Comment:
This doesn't ensure the latest user detail as the state could be of before an update event from another instance. It might be better to do this on the server side, else trigger a call to fetch the user detail before this and then get the state

Reply:
As the generated entity is in a microservice, I think it's best to trigger a call to fetch the user detail (I don't think sure the microservice would know where to call for the user info).

But I was thinking : we create a table both in the gateway and the microservice with the user information : how de we keep these tables synchronized ? Is it a good practice to duplicate these tables ?

Reply:
I don't like it very much either. I think we shouldn't have relationships to users in microservices.

Reply:
Yes, I don't see the point in allowing a relationship to an entity which is in another service as well, this also creates coupling. It's the same as a microservice having a relationship to an entity in another microservice. I thought we didn't allow this earlier in the days, I might be wrong. I would vote to limit this. If someone needs such a relationship it needs to be handled manually IMO

Reply:
I agree. Let's close this PR if everyone is ok.

ping @mraible as he was the one raising the issue in #8920

Reply:
> As the generated entity is in a microservice, I think it's best to trigger a call to fetch the user detail (I don't think sure the microservice would know where to call for the user info).

micro-service should get logged-in user details from Oauth2 token claims. Since, user details are managed on the IDP, we should limit data stored in gateway / micro-services to avoid sync issues.